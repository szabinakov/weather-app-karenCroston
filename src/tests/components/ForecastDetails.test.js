import React from "react";
import { render } from "@testing-library/react";
import ForecastDetails from "../../components/ForecastDetails";

describe("ForecastDetails", () => {
  const validProps = {
    date: 1111111,
    humidity: 30,
    temperature: {
      max: 22,
      min: 12,
    },
    wind: {
      speed: 13,
      direction: "s",
    },
  };

  it("renders correctly", () => {
    const { asFragment } = render(<ForecastDetails forecast={validProps} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders correct values to props", () => {
    const { getByText } = render(<ForecastDetails forecast={validProps} />);

    expect(getByText("Thu Jan 01 1970")).toHaveClass("forecast-details__date");
    expect(getByText("Max Temperature: 22°C")).toHaveClass(
      "forecast-details__temperature-max"
    );
    expect(getByText("Min Temperature: 12°C")).toHaveClass(
      "forecast-details__temperature-min"
    );
    expect(getByText("Humidity: 30")).toHaveClass("forecast-details__humidity");
    expect(getByText("Wind Speed: 13")).toHaveClass(
      "forecast-details__wind-speed"
    );
    expect(getByText("Wind Direction: s")).toHaveClass(
      "forecast-details__wind-direction"
    );
  });
});
