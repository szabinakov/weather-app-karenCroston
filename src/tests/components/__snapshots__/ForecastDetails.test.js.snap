// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastDetails renders correctly 1`] = `
<DocumentFragment>
  <div
    class="forecast-details"
  >
    <div
      class="forecast-details__date"
    >
      Thu Jan 01 1970
    </div>
    <div
      class="forecast-details__temperature-max"
    >
      Max Temperature: 22°C
    </div>
    <div
      class="forecast-details__temperature-min"
    >
      Min Temperature: 12°C
    </div>
    <div
      class="forecast-details__humidity"
    >
      Humidity: 30
    </div>
    <div
      class="forecast-details__wind-speed"
    >
      Wind Speed: 13
    </div>
    <div
      class="forecast-details__wind-direction"
    >
      Wind Direction: s
    </div>
  </div>
</DocumentFragment>
`;
